oximeter/collector/src/agent.rs

```diff
@@ -656,7 +928,10 @@ impl OximeterAgent {
         //
         // NOTE: This can either mean that the collection completed
         // successfully, or an error occurred in the collection pathway.
-        futures::future::join_all(collection_oneshots).await;
+        futures::future::try_join_all(collection_oneshots)
+            .await
+            .map(|_| ())
+            .map_err(|_| ForcedCollectionQueueFull)
```

Comment:
Do we need a `.map_err` here? I thought the oneshots would get back `ForcedCollectionQueueFull` if that's why they failed.

Reply:
I switched this from `join_all` to `try_join_all`, which means this is now fallible. I'm not sure that's what we want though, after looking at it again. I want to drive all the futures to completion, and return this error if any one of them failed. `try_join_all` will cancel all the others as soon as it finds an error. I'll rework this a bit to make sure we're doing the right thing here.

Reply:
I reworked this in 69dfa7a9d. We now return an error enum, rather than an empty struct, to distinguish between failing to send a force-collection request to the collection task because it was full; and failing to send or receive the result because the channel was closed.